### PR TITLE
Feature/font

### PR DESCRIPTION
**Pull request by jperezgonzal**
## Description
Se migró el módulo Font desde _android-ui_  a _andes_. 
Tener en cuenta que es necesario que la lib de _andes_ tiene que publicarse antes que _android-ui_ y _config-provider_ ya que estos dos dependen de los cambios que se realizaron.

## Checks
Are you sure you followed all those steps? In such case, let's say with me "I solemnly declare that ...":
- [ ] I have coded an example inside the Demo App,
- [ ] I have added proper tests,
- [ ] I have modified the [Changelog](https://github.com/mercadolibre/fury_andesui-android/blob/develop/CHANGELOG.md) file,
- [ ] I have updated GitHub wiki,
- [ ] I have the explicit approval of one or more members of the UX Team,
- [ ] I have updated the version in gradle.properties file.

## Change type
This is easy, let us know what this code is about:
- [x] This code adds a new component
- [ ] This code includes improvements to an existent component
- [ ] This code improves or modifies ONLY the Demo App.

## Check common errors
Whether you are an Android or iOS developer and if you're still there then we'll give you a present:
https://proandroiddev.com/how-to-maximize-androids-ui-reusability-5-common-mistakes-cb2571216a9f

### CODE REVIEW

**Comment by furyGorgoryAdmin** [comment]
### Informative Menu <h2>  ### 

### Use any of these tags in a pull request description or leave a comment ### 
| Tag | Description |
|---|---|
| **[canary]**| Creates a build that can only be deployed with canary strategy. | 
| **[build]** |  Creates version or build from a branch that has been tagged as "no releasable". | 
| **[name:version-name]** |Specify the name of the build. For example: [build] [name:1.2.3-name] or [canary] [name:1.2.3-name].|
| **[major]** | To automatically increase the first position of the version. For example: [build] [major] or [canary] [major].|
| **[minor]** | To automatically increase the second position of the version. For example: [build] [minor] or [canary] [minor]. |
| **[patch]** | To automatically increase the third position of the version. For example: [build] [patch] or [canary] [patch].|  

[For more information about ci-cd](http://docs.furycloud.io/?lang=esp&section=businessassurance)
[For more information about semver](https://semver.org/#summary)

File: components/src/main/java/com/mercadolibre/android/andesui/font/CalligraphyTypefaceSetter.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import android.graphics.Typeface
+import android.widget.Switch
+import android.widget.TextView
+import com.mercadolibre.android.andesui.font.TypefaceHelper.TypefaceSetter
+import uk.co.chrisjenx.calligraphy.TypefaceUtils
+
+/**
+ * This TypefaceSetter implements Calligraphy library.
+ */
+class CalligraphyTypefaceSetter : TypefaceSetter {
```
Comment (barriosnahuel) [inline components/src/main/java/com/mercadolibre/android/andesui/font/CalligraphyTypefaceSetter.kt:14]: no deberíamos nombrarlo más como calligraphy me parece

File: build.gradle [hunk 2]
```diff
@@ -23,6 +24,7 @@ wrapper {
 
 allprojects {
     repositories {
+        mavenLocal()
```
Comment (jiurchuk) [inline build.gradle:27]: Estamos seguros de que queremos tener esto fijo?

File: components/src/main/java/com/mercadolibre/android/andesui/font/Font.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    object FontConfig {
+        /* default */ var fonts: EnumMap<Font, String>? = null
+            private set
+
+        /**
+         * Sets the fonts to use
+         * @param fontsMap with Font as Key and String with path to the font as value
+         */
+
```
Comment (jiurchuk) [inline components/src/main/java/com/mercadolibre/android/andesui/font/Font.kt:42]: Empty line

File: components/src/main/java/com/mercadolibre/android/andesui/font/TypefaceSpanCompat.kt [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        }
+        paint.typeface = typeFace
+    }
+
+    private fun getOldStyle(oldTypeface: Typeface?): Int {
+        return oldTypeface?.style ?: 0
+    }
+
+    companion object {
+        private const val SKEW_X = -0.25f
```
Comment (jiurchuk) [inline components/src/main/java/com/mercadolibre/android/andesui/font/TypefaceSpanCompat.kt:46]: Qué representa este número?

File: components/src/main/java/com/mercadolibre/android/andesui/font/TypefaceHelper.kt [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+     * @param <T>   A generic for the textview
+     * @param view  The view to which apply the font
+     * @param font  The [Font] the text should have
+    </T> */
+    @JvmStatic
+    fun <T : TextView?> setTypeface(view: T, font: Font) {
+        if (this::typefaceSetter.isInitialized) {
+            typefaceSetter.setTypeface(view, font)
+        } else {
+            // Do log
```
Comment (lautimfernandez) [inline components/src/main/java/com/mercadolibre/android/andesui/font/TypefaceHelper.kt:36]: No nos olvidemos de esto

File: components/src/main/java/com/mercadolibre/android/andesui/font/TypefaceHelper.kt [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+     * @param context A context to obtain the font
+     * @param paint   The paint to which apply the font
+     * @param font    The [Font] the text should have
+     */
+    @JvmStatic
+    fun setTypeface(context: Context, paint: Paint, font: Font) {
+        if (this::typefaceSetter.isInitialized) {
+            typefaceSetter.setTypeface(context, paint, font)
+        } else {
+            // Do log
```
Comment (lautimfernandez) [inline components/src/main/java/com/mercadolibre/android/andesui/font/TypefaceHelper.kt:52]: Mismo que arriba

File: components/src/main/java/com/mercadolibre/android/andesui/font/TypefaceHelper.kt [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+     * @param context to use
+     * @param font to use
+     * @return associated typeface
+     */
+    @JvmStatic
+    fun getFontTypeface(context: Context, font: Font): Typeface? {
+        return if (this::typefaceSetter.isInitialized) {
+            typefaceSetter.getTypeface(context, font)
+        } else {
+            // Do log
```
Comment (lautimfernandez) [inline components/src/main/java/com/mercadolibre/android/andesui/font/TypefaceHelper.kt:68]: same

**Pull request closed without merging**